src/Symfony/Component/HttpKernel/DependencyInjection/RegisterControllerArgumentLocatorsPass.php

```diff
@@ -127,6 +127,11 @@ public function process(ContainerBuilder $container)
                     $type = ltrim($target = (string) ProxyHelper::getTypeHint($r, $p), '\\');
                     $invalidBehavior = ContainerInterface::IGNORE_ON_INVALID_REFERENCE;
 
+                    if (enum_exists($type)) {
```

Comment:
```suggestion
                    if (function_exists('enum_exists') && enum_exists($type)) {
```

Reply:
The php81, polyfill was here for this purpose but Nicolas came with a better suggestion :)